File: API/models/pedido.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class PedidoBase(BaseModel):
    folio: Optional[str] = None
    usuario_id: int
    estatus_id: int
    total: float = 0.00
    direccion_envio_id: int
    cliente_id: Optional[int] = None
    motivo_cancelacion: Optional[str] = None
    fecha_cancelacion: Optional[datetime] = None

File: Laravel/app/Models/Macuin/Pedido.php
<?php

namespace App\Models\Macuin;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Pedido extends Model
{
    protected $table = 'pedidos';

    public $timestamps = false;

    protected $casts = [
        'total' => 'decimal:2',
        'fecha_pedido' => 'datetime',
        'fecha_cancelacion' => 'datetime',
    ];

    protected $fillable = [
        'folio',
        'usuario_id',
        'estatus_id',
        'total',
        'direccion_envio_id',
        'cliente_id',
        'motivo_cancelacion',
        'fecha_cancelacion',
    ];

    public function cliente(): BelongsTo
    {
        return $this->belongsTo(Cliente::class, 'cliente_id');
    }

    public function estatus(): BelongsTo
    {
        return $this->belongsTo(EstatusPedido::class, 'estatus_id');
    }

    public function detalles(): HasMany
    {
        return $this->hasMany(DetallePedido::class, 'pedido_id');
    }
}
